### ISSUE TITLE
TypeScript: Function declaration not wrapped with nullable parameter

### ISSUE THREAD

**Issue by KingHenne** [prettier/prettier#2002]
**Input**

```ts
export function aPrettyLongFunction(aRatherLongParamName: string | null): string[] {}
```

**Expected Output**

```ts
export function aPrettyLongFunction(
  aRatherLongParamName: string | null
): string[] {}
```

**Actual Output**

```ts
export function aPrettyLongFunction(aRatherLongParamName: string | null): string[] {}
```

([interactive example](https://prettier.github.io/prettier/#%7B%22content%22%3A%22export%20function%20aPrettyLongFunction(aRatherLongParamName%3A%20string%20%7C%20null)%3A%20string%5B%5D%20%7B%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D))

---

Note that wrapping works when you replace `null` with `number` for example:

**Input**

```ts
export function aPrettyLongFunction(aRatherLongParamName: string | number): string[] {}
```

**Correct Output**

```ts
export function aPrettyLongFunction(
  aRatherLongParamName: string | number
): string[] {}
```

([interactive example](https://prettier.github.io/prettier/#%7B%22content%22%3A%22export%20function%20aPrettyLongFunction(aRatherLongParamName%3A%20string%20%7C%20number)%3A%20string%5B%5D%20%7B%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D))

**Comment by vjeux** [comment]
Wow, this is weird! Thanks for the report, we should definitely fix that. I'm not really sure why it's happening.

**Comment by vjeux** [comment]
I just published 1.4.3 which fixes it. Thanks for reporting it!
